cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -1189,6 +1189,25 @@ if (uring)
   endif()
 endif()
 
+#---Check for DAOS----------------------------------------------------------------
+if (daos AND daos_mock)
+  message(FATAL_ERROR "Options `daos` and `daos_mock` are mutually exclusive; only one of them should be specified.")
+endif()
+if (daos OR daos_mock)
+  message(STATUS "Looking for DAOS")
+  if(fail-on-missing)
+    find_package(DAOS REQUIRED)
+    find_package(libuuid REQUIRED)
```

Comment:
It will possibly make to fail LCG builds, since they are built with fail-on-missing always on...
@Axel-Naumann do you have an opinion on what to do here? to drop fail-on-missing or communicate this feature with LCG

Reply:
Wouldn't it fail only if the build request (and is not getting) the build of the DOAS based library?

Reply:
> ROOT has `class TUUID`. Can we use this instead of libuuid?

`libdaos` requires the type `uuid_t`, as defined by libuuid's `uuid.h`.  Anyway, we are not introducing this dependency (it was already required by Davix, I think).

Reply:
> It will possibly make to fail LCG builds, since they are built with fail-on-missing always on...
> @Axel-Naumann do you have an opinion on what to do here? to drop fail-on-missing or communicate this feature with LCG..

`daos` is not enabled by default.  Therefore, unless enabled, it shouldn't make any fail-on-missing build to fail.

Reply:
@jalopezg-r00t I am still confused by:
```
if (daos OR daos_mock)
    ...
    find_package(DAOS REQUIRED)
```
Is the DAOS library really required for `daos_mock`?

Reply:
> Is the DAOS library really required for `daos_mock`?

No, it is not.  I have fixed that, but still not pushed it (until all the other changes are done).